### ISSUE TITLE
Feature request: Return response and status code from APIGatewayRestResolver routes

### ISSUE THREAD

**Issue by theoutsider24** [awslabs/aws-lambda-powertools-python#1845]
### Use case

APIGatewayRestResolver currently supports 2 return values - either a Response object or a dict (see [_to_response](https://github.com/awslabs/aws-lambda-powertools-python/blob/d4feaedd42339899840b52ce61dd89d238fe7651/aws_lambda_powertools/event_handler/api_gateway.py#L731))

However, when returning a dict, the status code is hardcoded to 200

It is a common use case to want to return other status codes, eg. a 201, alongside a dict of the created object

### Solution/User Experience

Flask handles this by allowing a tuple to be returned with the first value being the response dict and the second being a status code (they also allow returning headers but I'm not pushing for that)

I currently use a subclassed resolver to add this functionality with a _to_response like this:
```
def _to_response(self, result: Union[Dict, Response, Tuple]) -> Response:
        if isinstance(result, tuple):
            response, code = result
            return Response(
                status_code=code,
                content_type=content_types.APPLICATION_JSON,
                body=self._json_dump(response),
            )
        else:
            return super()._to_response(result)
```

The flask implementation looks like this with additional validation and the header handling
```
  if isinstance(rv, tuple):
            len_rv = len(rv)

            # a 3-tuple is unpacked directly
            if len_rv == 3:
                rv, status, headers = rv  # type: ignore[misc]
            # decide if a 2-tuple has status or headers
            elif len_rv == 2:
                if isinstance(rv[1], (Headers, dict, tuple, list)):
                    rv, headers = rv
                else:
                    rv, status = rv  # type: ignore[assignment,misc]
            # other sized tuples are not allowed
            else:
                raise TypeError(
                    "The view function did not return a valid response tuple."
                    " The tuple must have the form (body, status, headers),"
                    " (body, status), or (body, headers)."
                )
```

Implementing this would allow something like this:
```
@app.post("/entity/")
def create_entity():
    data = router.current_event.json_body
    new_entity = Entity(**data)
    return new_entity.asdict(), 201
```

### Acknowledgment

- [X] This feature request meets [Lambda Powertools Tenets](https://awslabs.github.io/aws-lambda-powertools-python/latest/#tenets)
- [ ] Should this be considered in other Lambda Powertools languages? i.e. [Java](https://github.com/awslabs/aws-lambda-powertools-java/), [TypeScript](https://github.com/awslabs/aws-lambda-powertools-typescript/)

**Comment by theoutsider24** [comment]
An example of this change is shown here https://github.com/awslabs/aws-lambda-powertools-python/compare/develop...theoutsider24:aws-lambda-powertools-python:patch-1

**Comment by leandrodamascena** [comment]
Hi @theoutsider24. Thank you for opening this issue.

We have the Response class to handle cases where you need to return status_code, headers, cookies and other attributes with values different from the defaults. Flask also has a Response object that can be used like ours and has the same behavior as our Response class.

I understand the motivation of this issue and I will talk internally with the team about a decision for this. We need to check for possible break changes and to have the same experience as Flask, we should have the same types of responses that Flask has, which can be:

Body only: `return 'Powertools'`
Body and status code: `return 'Powertools', 200`
Body and headers: `return 'Powertools', {'App': 'Powertools'}`
Body, status code and header: `return 'Powertools', 200, {'App': 'Powertools'}`

As soon as I have a feedback I'll come back here to update this issue.

Thank you very much.

**Comment by heitorlessa** [comment]
agree is a good UX via a tuple for simple responses.

> they also allow returning headers but I'm not pushing for that

Yes, I'd be against supporting a third value (headers) - it defeats the UX ergonomics this proposes.

Response class is well suited for that instead.

**Comment by leandrodamascena** [comment]
So let's implement it this way:

Body only: `return 'Powertools' `
or
Body and status code: `return 'Powertools', 200`

Do you have bandwidth to send a PR @theoutsider24? It could be awesome to have your contribution on this project. I read your code and it’s looks pretty well, just need to write tests and a small change in this line https://github.com/awslabs/aws-lambda-powertools-python/compare/develop...theoutsider24:aws-lambda-powertools-python:patch-1#diff-bdb4b43087f89fdf398381f3d88048f6182562ca67e8907259d5c56d658465c1R744 (line 744) to change `len(tuple)` to `len(result)`.

**Comment by theoutsider24** [comment]
@leandrodamascena Yeah I could put together a PR, that implementation was very hastily put together 😂 
I'll get some testing in place for it

One thing to note is that `return 'Powertools'` is not explicitly supported - it may actually work as a return value since a json dumps of a string returns another string but the signature of the _to_response function, and the "APPLICATION_JSON" content type applied, do not seem prepared to accept a string
We could extend the type hinting to include str, as well as changing the content type to TEXT_PLAIN, however that could impact anyone who's currently using it in that way

**Comment by leandrodamascena** [comment]
> One thing to note is that `return 'Powertools'` is not explicitly supported - it may actually work as a return value since a json dumps of a string returns another string but the signature of the _to_response function, and the "APPLICATION_JSON" content type applied, do not seem prepared to accept a string

Yes, you are 100% right.. It was just a typo, consider a JSON as the actual code.

**Comment by theoutsider24** [comment]
Thanks @leandrodamascena, I've opened a PR https://github.com/awslabs/aws-lambda-powertools-python/pull/1853

**Comment by leandrodamascena** [comment]
This is now released under [2.7.0](https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v2.7.0) version!
